com.unity.netcode.gameobjects/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -678,17 +685,23 @@ internal void DestroySceneObjects()
         internal void ServerSpawnSceneObjectsOnStartSweep()
         {
             var networkObjects = UnityEngine.Object.FindObjectsOfType<NetworkObject>();
+            var networkObjectsToSpawn = new List<NetworkObject>();
 
             for (int i = 0; i < networkObjects.Length; i++)
             {
                 if (networkObjects[i].NetworkManager == NetworkManager)
                 {
                     if (networkObjects[i].IsSceneObject == null)
                     {
-                        SpawnNetworkObjectLocally(networkObjects[i], GetNetworkObjectId(), true, false, null, true);
+                        networkObjectsToSpawn.Add(networkObjects[i]);
                     }
                 }
             }
+
+            foreach (var networkObject in networkObjectsToSpawn)
+            {
+                SpawnNetworkObjectLocally(networkObject, GetNetworkObjectId(), true, false, null, true);
+            }
```

Comment:
this and the local variable are not needed? just move this call inside the for loop above?

Reply:
No, that's the whole point of the fix! :-)
The `for` enumerates all objects with `IsSceneObject` null. Those are the object we want to spawn.

But `SpawnNetworkObjectLocally` changes the children's `IsSceneObject`value. So, we can't start spawning until we're done enumerating. That's why those are split like this.

Reply:
> But SpawnNetworkObjectLocally changes the children's IsSceneObjectvalue.
ewwwww. gosh, okay.